jax/scipy/linalg.py

```diff
@@ -264,3 +263,120 @@ def tril(m, k=0):
 @_wraps(scipy.linalg.triu)
 def triu(m, k=0):
   return np.triu(m, k)
+
+@jit
+def expm(A):
+    """
+    Compute the matrix exponential using Pade approximation.
+    Based on TensorFlow's tf.linalg.expm
+
+    Args:
+    A : square matrix of type float32, float64, complex64, complex128
+    
+    Returns:
+    Matrix exponential of A , square matrix of same dimensions as A
+        
+
+    See: "The Scaling and Squaring Method for the Matrix
+    Exponential Revisited", Nicholas Higham, 2005.
+    
+    TODO: Write a version to exploit the sparsity of A
+    """
+    A = np.asarray(A)
+    if A.ndim != 2 or A.shape[0] != A.shape[1]:
+        raise ValueError('expected A to be a square matrix')
+    A_L1 = np_linalg.norm(A,1)
+    n_squarings = 0
+    def _nest_where(vals, cases):
+      assert len(vals) == len(cases) - 1
+      if len(vals) == 1:
+        return np.where(
+            np.less(A_L1, vals[0]), cases[0], cases[1])
+      else:
+        return np.where(
+            np.less(A_L1, vals[0]), cases[0],
+            _nest_where(vals[1:], cases[1:]))
```

Comment:
optional: you could probably make use `jax.numpy.select` to simplify this

Reply:
Yes. jax.numpy.select is more elegant.